File: backend/models/todo.py
from sqlmodel import SQLModel, Field, Relationship
from typing import Optional
from datetime import datetime

class TodoBase(SQLModel):
    title: str
    description: Optional[str] = None
    is_completed: bool = False

class Todo(TodoBase, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)
    user_id: int = Field(foreign_key="user.id")
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)

    # Relationship to user (using string reference to avoid circular import)
    user: "User" = Relationship(back_populates="todos")

class TodoCreate(TodoBase):
    pass

class TodoCreateWithUserId(TodoBase):
    user_id: int

class TodoUpdate(SQLModel):
    title: Optional[str] = None
    description: Optional[str] = None
    is_completed: Optional[bool] = None

class TodoRead(TodoBase):
    id: int
    created_at: datetime
    updated_at: datetime

File: frontend/types/todo.ts
// Todo entity types
export interface Todo {
  id: number;
  title: string;
  description: string;
  is_completed: boolean;
  due_date?: string | null;
  user_id: number;
  created_at: string;
  updated_at: string;
}

export interface TodoCreate {
  title: string;
  description: string;
  is_completed?: boolean;
  due_date?: string | null;
}

export interface TodoUpdate {
  title?: string;
  description?: string;
  is_completed?: boolean;
  due_date?: string | null;
}